config/v1/types_infrastructure.go

```diff
@@ -562,6 +573,23 @@ type BareMetalPlatformStatus struct {
 	// datacenter DNS, a DNS service is hosted as a static pod to serve those hostnames
 	// to the nodes in the cluster.
 	NodeDNSIP string `json:"nodeDNSIP,omitempty"`
+
+	// loadBalancerType defines if a user-managed or openshift-managed loadbalancer
+	// is to be used for the OpenShift API and Ingress endpoints.
+	// When set to OpenShiftManaged the static pods in charge of API and Ingress traffic load-balancing
+	// defined in the machine config operator will be deployed.
+	// When set to UserManaged these static pods will not be deployed and it is expected that
+	// the load balancer is configured out of band by the deployer.
+	// When omitted, this means no opinion and the platform is left to choose a reasonable default.
+	// The default value is OpenShiftManaged.
+	// +default="OpenShiftManaged"
+	// +kubebuilder:default:="OpenShiftManaged"
+	// +kubebuilder:validation:Enum:="OpenShiftManaged";"UserManaged"
+	// +kubebuilder:validation:Required
+	// +kubebuilder:validation:XValidation:rule="oldSelf == '' || self == oldSelf",message="loadBalancerType is immutable once set"
+	// +openshift:enable:FeatureSets=TechPreviewNoUpgrade
```

Comment:
Ooh, shiny. Got a docs link which describes exactly what this does?

Reply:
https://github.com/openshift/enhancements/blob/f7b697829e7856978c8b33baf374da74064dc363/enhancements/update/cvo-techpreview-manifests.md

Reply:
In terms of the CRD generator, compare the two manifests (default and techpreview), the new fields only exist in the techpreview version which means they will only be available when the CVO deploys the techpreview version of the CRD manifest